llarp/service/endpoint.cpp

```diff
@@ -316,57 +271,20 @@ namespace llarp
 #endif
 
       // deregister dead sessions
-      {
-        auto itr = m_DeadSessions.begin();
-        while(itr != m_DeadSessions.end())
-        {
-          if(itr->second->IsDone(now))
-            itr = m_DeadSessions.erase(itr);
-          else
-            ++itr;
-        }
-      }
+      EndpointUtil::DeregisterDeadSessions(now, m_DeadSessions);
       // tick remote sessions
-      {
-        auto itr = m_RemoteSessions.begin();
-        while(itr != m_RemoteSessions.end())
-        {
-          if(itr->second->Tick(now))
-          {
-            itr->second->Stop();
-            m_DeadSessions.emplace(itr->first, std::move(itr->second));
-            itr = m_RemoteSessions.erase(itr);
-          }
-          else
-            ++itr;
-        }
-      }
+      EndpointUtil::TickRemoteSessions(now, m_RemoteSessions, m_DeadSessions);
```

Comment:
I don't like this particularly. In C++17 this could just be:

```c++
m_DeadSessions.merge(EndpointUtil::TickRemoteSessions(now, m_RemoteSessions));
```

which I think would cleaner imo